lib/ansible/plugins/action/copy.py

```diff
@@ -407,12 +407,14 @@ def run(self, tmp=None, task_vars=None):
         local_follow = boolean(self._task.args.get('local_follow', True), strict=False)
 
         result['failed'] = True
-        if ((not source or source is None) and content is None) or dest is None:
-            result['msg'] = "src (or content) and dest are required"
-        elif source is not None and content is not None:
-            result['msg'] = "src and content are mutually exclusive"
+        if (not source or source is None) and content is None:
```

Comment:
If source is None, the conditional `not source` will be true. So I don't think the second part of this conditional will ever hit.